### DISCUSSION TITLE
Examples of `Cutlass / CuTe` in practice

### DISCUSSION THREAD

**Discussion by jeromeku** [NVIDIA/cutlass#1283] [Q&A]
What are good open source examples of `cutlass / cute` in practice?  

I'm aware of [Flash-Attention](https://github.com/Dao-AILab/flash-attention), [xformers](https://github.com/facebookresearch/xformers), pytorch (inductor alternative GEMM backend), [DeepSpeed](https://github.com/microsoft/DeepSpeed/tree/3e94f8c75116377d4b1c32b8c674368a27fb2a77/deepspeed/inference/v2/kernels/cutlass_ops), ...

**Comment by hwu36** [comment]
are you asking which oss projects are using cutlass?

**Reply by jeromeku** [reply]
@hwu36 
Yes -- OSS libraries that are using `Cutlass` and you would consider good usage patterns.

**Reply by hwu36** [reply]
@mnicely

**Reply by mnicely** [reply]
AITemplate - https://github.com/facebookincubator/AITemplate
Alibaba MNN - https://github.com/alibaba/MNN
TVM - https://github.com/apache/tvm
tiny-cuda-nn - https://github.com/NVlabs/tiny-cuda-nn
PaddlePaddle - https://github.com/PaddlePaddle/Paddle
FasterTransformer - https://github.com/NVIDIA/FasterTransformer

Just do a search for 'cutlass' through out Github and investigate the issues. You'll find many repos using CUTLASS
